src/Microsoft.Identity.Web.OWIN/AppBuilderExtension.cs

```diff
@@ -175,6 +175,23 @@ public static IAppBuilder AddMicrosoftIdentityWebApp(
 
                                 if (clientInfoFromServer != null && clientInfoFromServer.UniqueTenantIdentifier != null && clientInfoFromServer.UniqueObjectIdentifier != null)
                                 {
+                                    var uniqueTenantIdentifierClaim = context.AuthenticationTicket.Identity.FindFirst(c => c.Type == ClaimConstants.UniqueTenantIdentifier);
+                                    var uniqueObjectIdentifierClaim = context.AuthenticationTicket.Identity.FindFirst(c => c.Type == ClaimConstants.UniqueObjectIdentifier);
+                                    if (uniqueTenantIdentifierClaim != null && !string.Equals(clientInfoFromServer.UniqueTenantIdentifier, uniqueTenantIdentifierClaim.Value, StringComparison.OrdinalIgnoreCase))
```

Comment:
Why are you ignoring case? Shouldn't it be an exact match?

Reply:
This was per @bgavrilMS suggestion above. It was exact match before (Ordinal)

Reply:
This was @bgavrilMS suggestion as per above.

I went with an exact match as if the values differ something is not correct, but maybe @bgavrilMS can chime in on his thoughts

Reply:
I'm just thinking of supportability. Imagine a token with `uuid` and `Uuid` claims. But would like @jmprieur  to chime in on this, there is some other work happening around claims and case sensitivity.

Reply:
however right now i'm doing the compare on the value. should the compare be on the type and value, type only or value only?

Reply:
> however right now i'm doing the compare on the value. should the compare be on the type and value, type only or value only?

Both should be case-sensitive. The type comparison (claim lookup) is already case-sensitive which is what's referenced above. I see no reason for the value comparison to be case-_insensitive_ as I'd expect the value to be unchanged or else it indicates tampering with an internal claim.

Reply:
> I'm just thinking of supportability. Imagine a token with `uuid` and `Uuid` claims. But would like @jmprieur to chime in on this, there is some other work happening around claims and case sensitivity.

This is referring to the value comparison and not the type comparison.

Reply:
I think this is fine. AccountID in MSAL (as well as most string op are case insensitive).